src/sc4626.sol

```diff
@@ -57,4 +65,16 @@ abstract contract sc4626 is ERC4626, AccessControl {
         treasury = newTreasury;
         emit TreasuryUpdated(msg.sender, newTreasury);
     }
+
+    function _initiateFlashLoan() internal {
+        flashLoanInitiated = true;
+    }
+
+    function _finalizeFlashLoan() internal {
+        flashLoanInitiated = false;
+    }
+
+    function _isFlashLoanInitiated() internal view {
```

Comment:
this function can still revert altho you declare it `view`?

Reply:
seems unintuitive but i guess this is how it works, since you have the unit test

Reply:
yeah, the compiler suggested it to me.